streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java

```diff
@@ -274,6 +264,24 @@ public void deleteTopicsAndWait(final long timeoutMs, final String... topics) th
         }
     }
 
+    /**
+     * Deletes all topics and blocks until all topics got deleted.
+     *
+     * @param timeoutMs the max time to wait for the topics to be deleted (does not block if {@code <= 0})
+     */
+    public void deleteAllTopicsAndWait(final long timeoutMs) throws InterruptedException {
+        Set<String> topics = brokers[0].listTopics();
```

Comment:
nit: add `final`

Reply:
I'm a bit concerned using `listTopics` than using `JavaConverters.seqAsJavaListConverter(brokers[0].kafkaServer().zkClient().getAllTopicsInCluster()).asJava())` as only the latter can get the source-of-truth on ZK while the former may be subject to race conditions (e.g. if you create the topic and then call listTopics, it may not be included if the metadata was not propagated yet).

Reply:
This is a one-node cluster, though, right?

> /**
>  * Runs an in-memory, "embedded" Kafka cluster with 1 ZooKeeper instance and 1 Kafka broker.
>  */
> public class EmbeddedKafkaCluster extends ExternalResource {
>

Reply:
The javadoc is out dated: we actually create a `KafkaEmbedded[] brokers`.

Reply:
If there are race conditions, we need to fix them. Remember that Streams itself has to use AdminClient.